src/content/en/2019/javascript.md

```diff
@@ -0,0 +1,338 @@
+use `<link rel="preload"> `for scripts on their page.
+
+For prefetch:
+
+<insert graphic of metric 01_16>
+
+For both mobile and desktop, **0.08% (~0.1%)** of pages leverage prefetch for any of their
+scripts.
+
+# Newer APIs
+
+JavaScript continues to evolve as a language. A new version of the language standard itself, known
+as ECMAScript, is released every year with new APIs and features passing proposal stages to become a
+part of the language itself.
+
+With HTTP Archive, we can take a look at any newer API that is supported (or is about to be) and see
+how widespread its usage is. These APIs may already be used in browsers that support them _or_ with
+an accompanying polyfill to make sure they still work for every user.
+
+**How many sites use the following APIs?**
+
+-  [Atomics](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Atomics)
+-  [Intl](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl)
+-  [Proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy)
+-  [SharedArrayBuffer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer)
+-  [WeakMap](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakMap)
+-  [WeakSet](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakSet)
+
+<insert graphic of metric 01_18>
+
+Atomics (**0.38%**) and SharedArrayBuffer (**0.2%**) are barely visible on this chart since they are
+used on such few pages. Each of the other APIs here 
+
+It is important to note that the numbers here are approximations and they do not leverage
+[UseCounter](https://chromium.googlesource.com/chromium/src.git/+/master/docs/use_counter_wiki.md)
+to measure feature usage.
+
+# Source maps
+
+In many build systems, JavaScript files undergo minification to minimize its size and transpilation
+for newer language features that are not yet supported in many browsers. Moreover, language
+supersets like [TypeScript](https://www.typescriptlang.org/) compile to an output that can look
+extremely different from the original source code. For all these reasons, the final code served to
+the browser can unreadable and hard to decipher.
+
+A **source map** is an additional file accompanying a JavaScript file that allows a browser to map
+the final output to its original source. This can make debugging and analyzing production bundles
+much simpler.
+
+Although useful, there are a number of reasons why many sites may not want to include source maps in
+their final production site, such as choosing not to expose complete source code to the public. **So
+how many sites actually include sourcemaps?**
+
+<insert graphic of metric 01_25>
+
+For both desktop and mobile pages, the results are about the same. **17-18%** include a source map
+for at least one script on the page (detected as a first-party script with `sourceMappingURL`).
```

Comment:
AFAIK source maps can also be enabled via header (`SourceMap` or the old `X-SourceMap`). I don't think many websites include source maps this way, but it might be useful to check the stats.

Reply:
Ditto about new metrics